src/hooks/useChainSafeKeygen.ts

```diff
@@ -0,0 +1,70 @@
+import { SecretKey, PublicKey } from '@chainsafe/bls/herumi'
+import { deriveEth2ValidatorKeys, deriveKeyFromMnemonic } from '@chainsafe/bls-keygen'
+import { useState } from 'react'
+
+export type DeriveValidatorKeysReturnType = {
+  secretKey: SecretKey
+  publicKey: PublicKey
+}
+
+export type useChainSafeKeygenReturnType = {
+  isLoading: boolean
+  deriveValidatorKeys: (mnemonic: string, index: number) => Promise<DeriveValidatorKeysReturnType>
+  generatePubKey: (mnemonic: string, index: number) => Promise<string>
+}
+
+const useChainSafeKeygen = (): useChainSafeKeygenReturnType => {
+  const [isLoading, setLoading] = useState<boolean>(false)
+
+  const deriveValidatorKeys = async (
+    mnemonic: string,
+    index: number,
+  ): Promise<DeriveValidatorKeysReturnType> => {
+    if (index < 0) {
+      throw new Error('NON_NEGATIVE_NUMBER')
```

Comment:
In some places you use errors in `error.SomeThing` format and here all caps. Are these errors parsed somewhere? Otherwise just write the error in human-readable form and show it to the user. 
```suggestion
      throw new Error('pubkey index in derivation path must not be negative')
```

Reply:
yes these translation keys that are passed to a translation hook before being displayed